package.json

```diff
@@ -2,13 +2,44 @@
   "name": "algosdk",
   "version": "2.7.0",
   "description": "The official JavaScript SDK for Algorand",
+  "type": "commonjs",
   "main": "dist/cjs/index.js",
   "module": "dist/esm/index.js",
   "browser": {
     ".": "dist/browser/algosdk.min.js",
     "crypto": false
   },
   "types": "dist/types/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./dist/types/index.d.ts",
+      "import": "./dist/esm/index.js",
+      "require": "./dist/cjs/index.js"
+    },
+    "./dist/types": {
```

Comment:
Why/how did you decide to export these additional paths? Clients should not be importing anything besides the base library.

Reply:
It's purely to maintain compatibility with the current 2.x.x version, so as to not introduce any breaking changes. Because there is no exports defined, users can import anything. `dist` and `src` are both directories that are in the package, so technically can be imported by a user.

For example `algokit-utils-ts` uses the following import as it's not exported from an index. https://github.com/algorandfoundation/algokit-utils-ts/blob/b5ae35781796cbf2196916e83e0712229349ca4f/src/indexer-lookup.ts#L2.

Reply:
As maintainers, it's not practical for us to support these import paths. I'd much rather export the types that need to be exported through the main package than support these additional import paths.

Alternatively, I'd also be ok supporting the same export paths as in the 3.0.0 branch, though it may take some work to get that working on this branch

Reply:
I do agree, however you are already allowing users to import these paths, so are technically already supporting these.

The trouble I have is that if we do change the import paths, it is a breaking change and should be signalled as such via a major version bump. I'm assuming that isn't desirable either, due to the existing 3.0.0 branch?